java/java.hints/src/org/netbeans/modules/java/hints/errors/EnablePreview.java

```diff
@@ -0,0 +1,163 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.hints.errors;
+
+import com.sun.source.util.TreePath;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.SourceVersion;
+import org.netbeans.api.annotations.common.NonNull;
+import org.netbeans.api.java.platform.JavaPlatform;
+import org.netbeans.api.java.platform.JavaPlatformManager;
+import org.netbeans.api.java.queries.SourceLevelQuery;
+import org.netbeans.api.java.source.ClasspathInfo.PathKind;
+import org.netbeans.api.java.source.CompilationInfo;
+import org.netbeans.modules.java.hints.spi.ErrorRule;
+import org.netbeans.spi.editor.hints.ChangeInfo;
+import org.netbeans.spi.editor.hints.Fix;
+import org.openide.filesystems.FileObject;
+import org.openide.util.NbBundle;
+import org.openide.util.Parameters;
+import org.netbeans.modules.java.hints.spi.preview.PreviewEnabler;
+import org.netbeans.modules.java.hints.spi.preview.PreviewEnabler.Factory;
+import org.openide.modules.SpecificationVersion;
+import org.openide.util.Lookup;
+
+/**
+ * Handle error rule "compiler.err.preview.feature.disabled.plural" and provide
+ * the fix for Single Source Java File.
+ *
+ * @author Arunava Sinha
+ */
+public class EnablePreview implements ErrorRule<Void> {
+
+    private static final Set<String> ERROR_CODES = Collections.unmodifiableSet(new HashSet<String>(Arrays.asList(
+            "compiler.err.preview.feature.disabled",           //NOI18N
+            "compiler.err.preview.feature.disabled.plural",    // NOI18N
+            "compiler.err.is.preview")));                      // NOI18N
+
+    @Override
+    public Set<String> getCodes() {
+        return ERROR_CODES;
+    }
+
+    @Override
+    @NonNull
+    public List<Fix> run(CompilationInfo compilationInfo, String diagnosticKey, int offset, TreePath treePath, Data<Void> data) {
+        final FileObject file = compilationInfo.getFileObject();
+
+        if (file != null) {
+            SpecificationVersion platformVersion = null;
+
+            FileObject jlObject = compilationInfo.getClasspathInfo().getClassPath(PathKind.BOOT).findResource("java/lang/Object.class");
+            for (JavaPlatform platform : JavaPlatformManager.getDefault().getInstalledPlatforms()) {
+                if (jlObject == platform.getBootstrapLibraries().findResource("java/lang/Object.class")) {
+                    platformVersion = platform.getSpecification().getVersion();
+                    break;
+                }
+            }
```

Comment:
i am going to steal this for a different hint, since I needed to know the JDK version before and couldn't find any utility or anything in the lookup which would help there.